File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if (root == null) {
            return new ArrayList<>();
        }

        Queue<TreeNode> que = new ArrayDeque<>();
        List<Integer> result = new ArrayList<>();
        que.add(root);

        while (!que.isEmpty()) {
            int n = que.size();
            for (int i = 0; i < n; i++) {
                TreeNode node = que.poll();
                if (i == 0) {
                    result.add(node.val);
                }
                if (node.right != null) {
                    que.add(node.right);
                }
                if (node.left != null) {
                    que.add(node.left);
                }
            }
        }

        return result;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result;

        if (root == NULL) {
            return result;
        }

        queue<TreeNode*> que;
        que.push(root);

        while (!que.empty()) {
            int n = que.size();

            for (int i = 0; i < n; i++) {
                TreeNode* node = que.front();
                que.pop();

                if (i == 0) {
                    result.push_back(node->val);
                }
                if (node->right != NULL) {
                    que.push(node->right);
                }
                if (node->left != NULL) {
                    que.push(node->left);
                }
            }
        }

        return result;
    }
};